packages/core/core/prompt/attachments-to-parts.ts

```diff
@@ -0,0 +1,39 @@
+import { Attachment } from '@ai-sdk/ui-utils';
+import { ImagePart, TextPart } from './content-part';
+
+type ContentPart = TextPart | ImagePart;
+
+/**
+ * Converts a data URL of type text/* to a text string.
+ */
+export function dataUrlToText(dataUrl: string): string {
+  return atob(dataUrl.split(',')[1]);
+}
+
+/**
+ * Checks if a string is a data URL.
+ */
+function isDataURL(s: string): boolean {
+  return s.startsWith('data:');
```

Comment:
creating a URL object and then checking the protocol would be a safer check

Reply:
TIL, done!